### ISSUE TITLE
Is the following a valid "baseUri"

### ISSUE THREAD

**Issue by sichvoge** [raml-org/raml-spec#645]
Currently, the spec defines that the value of the `baseUri` node need to conform the RFC2396 specification (see [this issue](https://github.com/raml-org/raml-spec/issues/644) for updating it to RFC3986).

Both specs RFC2396(old) and RFC3986(new) clearly define that a valid URI needs to have the following syntax:

```
URI            = scheme ":" hier-part [ "?" query ] [ "#" fragment ]

hier-part   = "//" authority path-abempty
                      / path-absolute
                      / path-rootless
                      / path-empty
```

That means any URI without a `scheme` is invalid. 

The problem I have is that it's unclear if RAML actually follows this as well. No RAML parser complains about me putting in `baseUri: example.com`. 

Questions: 

* Is `baseUri: example.com` a valid value?
* What schemes do we support? Right now, nothing really prevents anyone from putting in 'ftp', 'file', or anything else.

**Comment by svacas** [comment]
both RFCs also cover [relative URI format](https://tools.ietf.org/html/rfc3986#section-4.2). For us it allows exposing the same RAML on different endpoints without need of rewriting the baseUri attribute.

**Comment by sichvoge** [comment]
If you look closely, they are talking about relative paths. So, for example, `/customers` is valid, but `example.com` or even `example.com/customers` would be invalid.

**Comment by svacas** [comment]
that's OK, I thought you were implying that `baseUri` must be absolute, as per the grammar snippet you pasted it.

**Comment by sichvoge** [comment]
No, my question really was about if `example.com` should be valid or not. At the moment, following the RAML spec and the way it defines the value of `baseUri`, it should not be valid.
